src/xunit.analyzers/AssertIsTypeShouldNotBeUsedForAbstractTypeFixer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Xunit.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class AssertIsTypeShouldNotBeUsedForAbstractTypeFixer : CodeFixProvider
+    {
+        private const string Title = "Use Assert." + IsAssignableFrom;
+
+        private const string IsAssignableFrom = "IsAssignableFrom";
+
+        private const string IsType = "IsType";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(Descriptors.X2017_AssertIsTypeShouldNotBeUsedForAbstractType.Id);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var invocation = root.FindNode(context.Span).FirstAncestorOrSelf<InvocationExpressionSyntax>();
+            var methodName = invocation.GetGenericName()?.Identifier.Text;
+            if (methodName == IsType)

```

Comment:
@marcind I am hesitant for `Assert.False(item is IInterface)` because I don't know that `is` has exactly the same semantics as `IsAssignableFrom`, and an `IsNotAssignableFrom` might come in the future-- that's why I said *if* any code fix was offered. However, your second point about refactoring a concrete class into an abstract one/an interface is valid, so you've convinced me to create a codefix that removes the assertion.